aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -30,6 +32,50 @@ struct RotationProofChallenge {
     new_public_key: Vec<u8>,
 }
 
+#[test]
+fn rotate_auth_key_ed25519_to_ed25519() {
+    let mut harness = MoveHarness::new();
+    let mut account1 = harness.new_account_with_key_pair();
+
+    let account2 = harness.new_account_with_key_pair();
+    // assert that the payload is successfully processed (the signatures are correct)
+    assert_successful_payload_key_rotation_from_ed25519_to_ed25519(
+        &mut harness,
+        account1.clone(),
+        *account1.address(),
+        10,
+        account2.privkey.clone(),
+        account2.pubkey.clone(),
+    );
+    // rotate account1's keypair to account2
+    account1.rotate_key(account2.privkey, account2.pubkey);
+    // verify that we can still get to account1's originating address
+    verify_originating_address(&mut harness, account1.auth_key(), *account1.address());
+}
+
+#[test]
+fn rotate_auth_key_ed25519_to_multi_ed25519() {
+    let mut harness = MoveHarness::new();
+    let account1 = harness.new_account_with_key_pair();
+
+    let private_key = MultiEd25519PrivateKey::generate_for_testing();
+    let public_key = MultiEd25519PublicKey::from(&private_key);
+    let auth_key = AuthenticationKey::multi_ed25519(&public_key);
+
+    // assert that the payload is successfully processed (the signatures are correct)
+    assert_successful_payload_key_rotation_from_ed25519_to_multi_ed25519(
+        &mut harness,
+        account1.clone(),
+        *account1.address(),
+        10,
+        private_key,
+        public_key,
+    );
+
```

Comment:
You don't call `rotate_key` here as you do in the other `rotate_auth_key_ed25519_to_ed25519 ` function...

Reply:
`rotate_key` only takes in ed25519 public key and private key as parameters (we cannot pass in multied25519 public key and private key in this case, but it's fine. it's a test function). for the other functions, especially the `rotate_auth_key_twice`, we are trying to rotate from the same account twice, so it's necessary to update the private key and public key after the first rotation using `rotate_key`. all the other test functions only rotate once, so we don't really need to do `rotate_key`